src/api/mod.rs

```diff
@@ -15,16 +17,24 @@ pub struct Client {
 }
 
 impl Client {
-    pub fn new(server_url: String, token: String, unsecured_certificate: bool, with_proxy: bool) -> Client {
+    pub fn new(
+        server_url: String,
+        token: String,
+        unsecured_certificate: bool,
+        with_proxy: bool,
+    ) -> Client {
         let mut http_client = ureq::AgentBuilder::new()
             .timeout_connect(Duration::from_secs(2))
             .timeout(Duration::from_secs(5))
             .user_agent(format!("openbas-implant/{}", VERSION).as_str())
             .try_proxy_from_env(with_proxy);
         if unsecured_certificate {
             let arc_crypto_provider = Arc::new(rustls::crypto::ring::default_provider());
-            let config = rustls_platform_verifier::tls_config_with_provider(arc_crypto_provider)
-                .expect("Failed to create TLS config with crypto provider");
+            let config = ClientConfig::builder_with_provider(arc_crypto_provider)
+                .with_safe_default_protocol_versions()
+                .unwrap()
```

Comment:
**question:** Can we keep the expect here ?

.expect("Failed to create TLS config with crypto provider")